.github/workflows/check-mainnet-config.yml

```diff
@@ -33,21 +33,52 @@ jobs:
     - name: Refresh cardano-node mainnet configuration
       run: |
         nix build --no-update-lock-file .#hydraJobs.cardano-deployment
-        SOURCE=$(pwd)/result
-        TARGET=$(pwd)/configuration/cardano
-
-        copyFile() {
-          echo $1
-          cp ${SOURCE}/$1 ${TARGET}/$1
-        }
-
-        copyFile "mainnet-alonzo-genesis.json"
-        copyFile "mainnet-byron-genesis.json"
-        copyFile "mainnet-conway-genesis.json"
-        copyFile "mainnet-config.json"
-        copyFile "mainnet-shelley-genesis.json"
-        copyFile "mainnet-topology.json"
-
-    - name: Check mainnet for mainnet configuration differences
+
+    - name: Check if configuration from nix is a superset of in-repo configuration
+      shell: python
       run: |
```

Comment:
This now checks if all configuration fields from repo configuration are present in the nix configuration and are equal. Extra fields in nix configuration do not make this check fail.

Reply:
We have a couple of python snippets in `cardano-node` already.
Anyway, I've rewritten this using `jq` - I hope it's better now.